### PR TITLE
CM-1531 notifications for posts

### PR DESCRIPTION
**Pull request by watchtower314**
Ticket: https://daostack1.atlassian.net/jira/software/projects/CM/boards/7?selectedIssue=CM-1531

### CODE REVIEW

**Comment by alexander2001i** [comment]
Love the labels!

File: packages/firebase/functions/src/notification/notification.ts [hunk 1]
```diff
@@ -334,6 +335,31 @@ export const notifyData: Record<string, IEventData> = {
       }
     )
   },
+  [EVENT_TYPES.DISCUSSION_CREATED]: {
+    // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+    data: async (discussionId: string) => {
+      const discussion = (await discussionDb.getDiscussion(discussionId));
+      const commonId = discussion.commonId
```
Comment (alexander2001i) [inline packages/firebase/functions/src/notification/notification.ts]: ;

File: packages/firebase/functions/src/notification/notification.ts [hunk 2]
```diff
@@ -334,6 +335,31 @@ export const notifyData: Record<string, IEventData> = {
...
+
+      const path = `Discussions/${commonId}/${discussionId}`
```
Comment (alexander2001i) [inline packages/firebase/functions/src/notification/notification.ts]: ;

**Review by alexander2001i — COMMENTED** [review]
I think that PR is made on top of the one before, so I did not spend a lot of time reviewing it. Only some missing commas (sorry for the lack of EsLint, will try to bring it ASAP)

**Comment by alexander2001i** [comment]
@watchtower314 The ESLint action should be working, just merge with `main`

File: packages/firebase/functions/src/common/business/updateCommon.ts [hunk 3]
```diff
@@ -1,22 +1,59 @@
...
+import { IUpdatableCommonEntity } from '../database/updateCommon';
 import { commonDb } from '../database';
-import { ICommonUpdate } from '../types';
 import { createEvent } from '../../util/db/eventDbService';
 import { EVENT_TYPES } from '../../event/event';
-import { commonEditHistoryDb } from '../../commonEditHistory/database';
+import { commonRuleValidationSchema } from '../../util/schemas';
+import { createCommonHistory } from '../../commonEditHistory/business';
+import { CommonError } from '../../util/errors';
+import {validate} from '../../util/validate';
```
Comment (alexander2001i) [inline packages/firebase/functions/src/common/business/updateCommon.ts:11]: Spot the difference 😄

File: packages/firebase/functions/src/common/business/updateCommon.ts [hunk 4]
```diff
@@ -1,22 +1,59 @@
...
+    .string()
+    .required(),
+  
+  userId: yup
+    .string()
+    .required(),
+
+  changes: yup.object({
+     name: yup.string().optional(),
+     rules: yup.array(commonRuleValidationSchema).optional(),
```
Comment (alexander2001i) [inline packages/firebase/functions/src/common/business/updateCommon.ts:27]: It is not a problem but you don't need to mark it as optional as it is optional by default fyi

**Review by alexander2001i — APPROVED** [review]

File: packages/firebase/functions/src/common/business/updateCommon.ts (hunk 3 above)
Comment (watchtower314) [reply packages/firebase/functions/src/common/business/updateCommon.ts:11]: yes I see it 😂 🤦🏽‍♀️

**Pull request merged**